jetty-websocket/websocket-jetty-tests/src/test/java/org/eclipse/jetty/websocket/tests/AnnotatedPartialListenerTest.java

```diff
@@ -0,0 +1,247 @@
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.message, is("hell"));
+        assertThat(segment.last, is(false));
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.message, is("o w"));
+        assertThat(segment.last, is(false));
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.message, is("orld"));
+        assertThat(segment.last, is(true));
+    }
+
+    @Test
+    public void testAnnotatedPartialByteBuffer() throws Exception
+    {
+        PartialByteBufferListener endpoint = new PartialByteBufferListener();
+        try (Session session = client.connect(endpoint, serverUri).get(5, TimeUnit.SECONDS))
+        {
+            session.getRemote().sendPartialBytes(BufferUtil.toBuffer("hell"), false);
+            session.getRemote().sendPartialBytes(BufferUtil.toBuffer("o w"), false);
+            session.getRemote().sendPartialBytes(BufferUtil.toBuffer("orld"), true);
+        }
+
+        PartialByteBufferListener.MessageSegment segment;
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.buffer, is(BufferUtil.toBuffer("hell")));
+        assertThat(segment.last, is(false));
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.buffer, is(BufferUtil.toBuffer("o w")));
+        assertThat(segment.last, is(false));
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.buffer, is(BufferUtil.toBuffer("orld")));
+        assertThat(segment.last, is(true));
+    }
+
+    @Test
+    public void testAnnotatedPartialByteArray() throws Exception
+    {
+        PartialByteArrayListener endpoint = new PartialByteArrayListener();
+        try (Session session = client.connect(endpoint, serverUri).get(5, TimeUnit.SECONDS))
+        {
+            session.getRemote().sendPartialBytes(BufferUtil.toBuffer("hell"), false);
+            session.getRemote().sendPartialBytes(BufferUtil.toBuffer("o w"), false);
+            session.getRemote().sendPartialBytes(BufferUtil.toBuffer("orld"), true);
+        }
+
+        PartialByteArrayListener.MessageSegment segment;
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.bytes, is("hell".getBytes()));
+        assertThat(segment.last, is(false));
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.bytes, is("o w".getBytes()));
+        assertThat(segment.last, is(false));
+
+        segment = Objects.requireNonNull(endpoint.messages.poll(5, TimeUnit.SECONDS));
+        assertThat(segment.bytes, is("orld".getBytes()));
+        assertThat(segment.last, is(true));
+    }
+}
```

Comment:
Please add a test that fails deployment if there are 2 `@OnWebSocketMessage`, one for full message, one for partial message.
It's probably covered elsewhere, but since this is a new feature, it's worth making sure the new feature is covered, so maybe you need to add/expand the tests where multiple annotations are tested.

Reply:
Done.